qiskit/transpiler/passes/optimize_phase_shift_gates.py

```diff
@@ -0,0 +1,120 @@
+warnings.filterwarnings('ignore')
+
+
+class OptimizePhaseShiftGates(TransformationPass):
+    """merge/cancel phase-shift gates in dag."""
+
+    def run(self, dag):
+        """
+        Run one pass of phase-shift gate optimization and cx cancellation
+        on the circuit. The function looks for the circuit configuration
+        with the lowest number of gates and circuit depth
+
+        Args:
+            dag (DAGCircuit): the directed acyclic graph to run on.
+        Returns:
+            DAGCircuit: Transformed DAG.
+        """
+        t_counter = {}  # Counts the cumulative phase of T-gates and other z-rotation gates
+        t_position_counter = {}  # Keeps track of the locations of T-gates and other z-rot. gates
+        state_tracker = np.zeros((dag.width(), dag.size())).astype('int')
+        for i in range(dag.width()):
+            state_tracker[i][i] = 1  # initial state
+        k = dag.width()  # number of initial variables
+        for node in dag.topological_nodes():
+            if node.type == 'op':
+                if node.name == 'cx':
+                    state_tracker[node.qargs[1][1]] ^= state_tracker[node.qargs[0][1]]
+                elif node.name in ['t', 'tdg', 's', 'sdg', 'z']:
+                    phase_t_gate_equivalent = {'t': 1, 'tdg': 7, 's': 2, 'sdg': 6, 'z': 4}[
+                        node.name]
+                    if str(state_tracker[node.qargs[0][1]]) in t_counter:
+                        t_counter[str(state_tracker[node.qargs[0][1]])] += phase_t_gate_equivalent
+                        t_position_counter[str(state_tracker[node.qargs[0][1]])].append(node)
+                    else:
+                        t_counter[str(state_tracker[node.qargs[0][1]])] = phase_t_gate_equivalent
+                        t_position_counter[str(state_tracker[node.qargs[0][1]])] = [node]
+                elif node.name == 'u1':
+                    if str(state_tracker[node.qargs[0][1]]) in t_counter:
+                        t_counter[str(state_tracker[node.qargs[0][1]])] += node.op.params[
+                            0] * 4 / np.pi
+                        t_position_counter[str(state_tracker[node.qargs[0][1]])].append(node)
+                    else:
+                        t_counter[str(state_tracker[node.qargs[0][1]])] = node.op.params[
+                            0] * 4 / np.pi
+                        t_position_counter[str(state_tracker[node.qargs[0][1]])] = [node]
+                else:
+                    state_tracker[node.qargs[0][1]] = np.zeros(dag.size())
+                    state_tracker[node.qargs[0][1]][k] = 1
+                    k += 1
+        circuit_length_min = dag.node_counter+1
+        circuit_depth_min = dag.depth() + 1
+        cx_cancel_pass = CXCancellation()
+        for key in t_counter:
+            if t_counter[key] % 8 == 0:
+                for i in t_position_counter[key]:
+                    dag.remove_op_node(dag._id_to_node[i._node_id])  # Remove cancelled nodes
```

Comment:
I think this can just be `dag.remove_op_node(i)`. It seems as though you are getting the id of the node, and then using this id to identify the node, but you could just use the node in the first place.